src/dns/local.rs

```diff
@@ -0,0 +1,190 @@
+use std::collections::HashMap;
+use std::net::IpAddr;
+use crate::dns::{DNSEntry, DnsResolver, IpEntry, ResolveType};
+use crate::types;
+use domain_lookup_tree::DomainLookupTree;
+use std::str::FromStr;
+use std::time;
+
+/// Local override table that can be used instead of using /etc/hosts or similar 3rd party dns system.
+pub struct LocalOverrideTable {
+    /// Entries in the local override table.
+    entries: HashMap<String, DNSEntry>,
+    /// Domaintree is a hierarchical lookup tree for quick scanning of (wildcard) domains
+    tree: DomainLookupTree,
```

Comment:
The wildcard matcher has been replaced by the tree lookup crate. It will match deeper wildcards better (*.example.org would match foo.bar.baz.example.org, and we don't want that).

Also, we now do not have to do a difficult process of matching both specific and wildcard matching.